.github/actions/create-version-tag/action.yml

```diff
@@ -0,0 +1,58 @@
+name: Create new version tag
+
+inputs:
+  last-version: 
+    type: string
+    required: true
+
+  version-strategy:
+    type: string
+    required: true
+
+  slack-webhook-url: 
+    type: string
+    required: true
+
+outputs:
+  next-version:
+    description: "The next version given the strategy prodivded. It has a 'v' preffix on it."
+    value: ${{ steps.next-version.outputs.version }}
+    
+runs:
+  using: "composite"
+  steps:
+    - name: Set package.json current version
+      shell: bash
+      run: |
+          # remove the "v" preffix.
+          current_version=$(echo "${{ inputs.last-version }}" | tr -d 'v')
+
+          # update the package.json file with the current version.
+          jq --arg current_version "$current_version" '.version = $current_version' package.json > tmp.json && mv tmp.json package.json
+
+    - name: Run npm version
+      shell: bash
+      run: "npm version ${{ inputs.version-strategy }} --git-tag-version=false --allow-same-version"
+
+    - name: Get next version
+      shell: bash
+      id: next-version
+      run: |
+        echo "version=v$(jq '.version' package.json --raw-output)" >> "$GITHUB_OUTPUT"
+
+    - name: Create release git tag
+      shell: bash
+      run: git tag -fa ${{ steps.next-version.outputs.version }} -m "${{ github.event.inputs.title }}"
+
+    - name: Push tag to main branch
+      shell: bash
+      # We use --force here to handle any already existing tags - In case we need to re-run this workflow.
```

Comment:
I thought about that. I tried to minimize possible developer errors that might break this workflow. In this case, the developer would forget about the new workflow and locally run a npm version command. This would guarantee that any wrongly created tag won't break this workflow.

Another upside I see with this approach is the ease of use to just re-run the main `Publish` workflow, in case the first step fails, instead of reaching into manual steps for everything else. The tag version is always based on the last release made, so if we're trying to create the same tag again, it's definitely because something went wrong, and we're trying to fix it.

But I might be missing some other edge cases that we might not want to use the `--force` flag. Wdyt?

Reply:
That makes sense! I think we're fine to use it. Let's leave it as is ✅

Reply:
Thanks for the explanation @guiquintelas !

> In this case, the developer would forget about the new workflow and locally run a npm version command.

They would have to push the created tag to main as well, right? And wouldn't the job override the existing tag in this case? In a way, it would be failing silently. There was something wrong with the tag, but the job fixed it without producing an error.

Would it make more sense to fail in this case? First of all, no one should push a tag to main. If they do, they need to remove it when they are aware (either on GH or locally via git). I think our git tags should always be "clean" and when something is wrong like a developer pushing a tag accidentally and running the job afterwards, I think the job should fail, explicitly call out the error, and someone needs to go to tags and clean it up before running the job again. 

-----------

Another scenario is, if I run the job with an existing version, would that mean it will override that tag? Again I think it might be safer to error out in this case. If someone is trying to push a version that already exists on NPM or in our git tags, that shouldn't be allowed or we would be re-writing history. 

(Let me know if I'm completely misunderstanding this)

Reply:
Totally agree @esezen.

I had the wrong mindset that the developer would just re-run the whole pipeline when encountering an error. That could cause many cascading errors. The best way is to run each subworkflow individually. And the best way to ensure that is to prevent the developer from re-running the main pipeline (supposing the error happened after the tag creation - most likely).

I imagine the two most probable actions to fail are:
1. Compiling the library.
2. Compiling the storybook.

The course of action would then be fixing the error and:
1. Manually re-running:
  a. Publish to NPM.
  b. Publish to CDN.
  c. Deploy storybook (if needed).
2. Just re-run the "Deploy storybook" action.

In both cases, we wouldn't be able to re-run the main pipeline because the GH release would have already been created. 

> I think the job should fail, explicitly call out the error, and someone needs to go to tags and clean it up before running the job again.

This 👌 

Thanks for shining a light on this @esezen!